src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/20.diff.sql

```diff
@@ -0,0 +1,353 @@
+        DECLARE @timestamp datetime2(7) = DATEADD(hour, DATEDIFF(hour, 0, sysutcdatetime()), 0);
+        
+        /* Ensures the next boundary value is greater than the current datetime. */
+        IF (@rightPartitionBoundary < @timestamp) BEGIN
+            SET @rightPartitionBoundary = @timestamp;
+        END;
+                            
+        /* Adds one hour for the next partition. */
+        SET @rightPartitionBoundary = DATEADD(hour, 1, @rightPartitionBoundary);
+        
+        /* Creates new empty partition by creating new boundary value and specifying NEXT USED file group. */
+        ALTER PARTITION SCHEME PartitionScheme_ResourceChangeData_Timestamp NEXT USED [Primary];
+        ALTER PARTITION FUNCTION PartitionFunction_ResourceChangeData_Timestamp() SPLIT RANGE(@rightPartitionBoundary);
+        
+        SET @partitionBoundary = @rightPartitionBoundary;
+
+    COMMIT TRANSACTION
+END;
+GO 
+
+--
+-- STORED PROCEDURE
+--     FetchResourceChanges_2
+--
+-- DESCRIPTION
+--     Returns the number of resource change records from startId. The start id is inclusive.
+--
+-- PARAMETERS
+--     @startId
+--         * The start id of resource change records to fetch.
+--     @lastProcessedDateTime
+--         * The last checkpoint datetime.
+--     @pageSize
+--         * The page size for fetching resource change records.
+--
+-- RETURN VALUE
+--     Resource change data rows.
+--
+CREATE OR ALTER PROCEDURE dbo.FetchResourceChanges_2
+    @startId bigint,
+    @lastProcessedDateTime datetime2(7),
+    @pageSize smallint
+AS
+BEGIN
+
+    SET NOCOUNT ON;
+    
+    /* Given the fact that Read Committed Snapshot isolation level is enabled on the FHIR database,
+       using the Repeatable Read isolation level table hint to avoid skipping resource changes
+       due to interleaved transactions on the resource change data table. 
+       In Repeatable Read, the select query execution will be blocked until other open transactions are completed
+       for rows that match the search condition of the select statement. 
+       A write transaction (update/delete) on the rows that match 
+       the search condition of the select statement will wait until the current transaction is completed. 
+       Other transactions can insert new rows that match the search conditions of statements issued by the current transaction.
+       But, other transactions will be blocked to insert new rows during the execution of the select query, 
+       and wait until the execution completes. */
+    SELECT TOP(@pageSize) Id,
+      Timestamp,
+      ResourceId,
+      ResourceTypeId,
+      ResourceVersion,
+      ResourceChangeTypeId
+      FROM dbo.ResourceChangeData WITH (REPEATABLEREAD)
+    WHERE Timestamp >= @lastProcessedDateTime AND Id >= @startId 
+    ORDER BY Timestamp ASC, Id ASC;
+END;
+GO
+
+EXEC dbo.LogSchemaMigrationProgress 'Creating partition function, and scheme for resource change data table.';
```

Comment:
Could we be more specific
Creating PartitionFunction_ResourceChangeData_Timestamp and PartitionScheme_ResourceChangeData_Timestamp.

Reply:
Updated the log.